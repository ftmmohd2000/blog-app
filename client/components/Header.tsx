import React from "react";

interface IHeaderProps {
  title: string;
}

export const Header = ({ title }: IHeaderProps) => {
  return (
    <div>
      <h1>Blog app</h1>
      <h2>{title}</h2>
    </div>
  );
};
