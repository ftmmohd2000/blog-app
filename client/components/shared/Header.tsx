import React from "react";

interface IHeaderProps {
  title: string;
}

export const Header = ({ title }: IHeaderProps) => {
  return (
    <div className="jumbotron">
      <h1>Blog app</h1>
      <h2>{title}</h2>
    </div>
  );
};
