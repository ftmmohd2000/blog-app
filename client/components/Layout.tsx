import React from "react";

import Head from "next/head";
import { Header } from "./Header";

interface ILayoutProps {
  children: (Element | JSX.Element)[] | Element | JSX.Element;
  pageTitle: string;
}

export const Layout = ({ children, pageTitle }: ILayoutProps) => {
  return (
    <div className="container">
      <Header title={pageTitle} />
      <Head>
        <title>{pageTitle}</title>
      </Head>
      {children}
    </div>
  );
};
