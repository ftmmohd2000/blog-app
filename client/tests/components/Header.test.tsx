import React from "react";
import { ShallowWrapper, shallow } from "enzyme";
import { Header } from "../../components/Header";

let wrapper: ShallowWrapper;
let title: string;

beforeEach(() => {
  title = "Some Title";
  wrapper = shallow(<Header title={title} />);
});

test("should render Header correctly", () => {
  expect(wrapper).toMatchSnapshot();
});
