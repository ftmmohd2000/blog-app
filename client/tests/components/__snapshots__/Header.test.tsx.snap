// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<div
  className="jumbotron"
>
  <h1>
    Blog app
  </h1>
  <h2>
    Some Title
  </h2>
</div>
`;
